packages/adapter/lib/adapter/adapter.js

```diff
@@ -1393,7 +1394,7 @@ function Adapter(options) {
         }
     };
 
-    const createInstancesObjects = (instanceObj, callback) => {
+    const createInstancesObjects = async (instanceObj, callback) => {
```

Comment:
Oh wow - how was this not noticed before?

Reply:
because the async calls were in a foreach callback and so kind of all run parallel ...